src/components/FormFeedback.js

```diff
@@ -1,35 +1,125 @@
 import React, { Component } from "react";
+import { clone } from "lodash";
+
+const rules = {
+  'email': (value) => {
+    if (!value || /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(value)) return false;
+    return 'Please enter a valid email.';
+  },
+  'required': (value) => {
+    if (typeof value !== 'undefined' && value !== null && /^\S/.test(value)) return false;
+    return 'This field is required.';
+  }
+};
+
+const formConfig = {
+  'stage': {
+    1: ['site-feedback-options'],
+    2: ['site-feedback-textarea', 'site-feedback-email']
+  },
+  'fieldRules' : {
+    'site-feedback-options': ['required'],
+    'site-feedback-textarea': ['required'],
+    'site-feedback-email': ['email']
+  }
+};
 
 class FormFeedback extends Component {
 
   constructor(props) {
     super(props);
     this.state = {
       stage: 1,
+      values: {},
+      errors: {}
     };
   }
 
-  setStage = (stage) => {
+  handleFieldChange = (e) => {
+
+    const values = clone(this.state.values);
```

Comment:
This is the es6 way to do this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
I do like the simplicity of the lodash syntax. Either way you do it, copy/clone is the way to go. 👏 
https://stackoverflow.com/questions/29537299/react-how-do-i-update-state-item1-on-setstate-with-jsfiddle#38378350

Reply:
I tend to like lodash for it's clarity -- i know what clone means, but Object.assign is a little esoteric. But eh, one less include, so I'll update

Reply:
yeah, I go back and forth. My impulse is to use what JS gives you out of the box. But I also prefer clean reading and lodash/underscore are so well documented and widely used that they might as well be core JS. Really meant to just share with you that its now possible to do in ES6, but I'd leave it to chef's choice on which function to use in the future.